### PR TITLE
Refactor/use slog library

### PR DESCRIPTION
**Pull request by jluzny**
This PR replaces all usages of _log_ and _Printx_ functions with the new standard _slog_ library calls (Info, Warn, Error). 
Almost all of the changes are very straightforward 1:1 replacements, so I assume there are no breaking changes.

Moreover example_live_test is extended:
- new test of daily schedules 
- uses devslog library for pretty printing into consolej

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ you can use to register schedules and listeners.
 */
 func NewApp(request NewAppRequest) (*App, error) {
 	if request.IpAddress == "" || request.HAAuthToken == "" || request.HomeZoneEntityId == "" {
-		log.Fatalln("IpAddress, HAAuthToken, and HomeZoneEntityId are all required arguments in NewAppRequest.")
+		slog.Error("IpAddress, HAAuthToken, and HomeZoneEntityId are all required arguments in NewAppRequest.")
```
Comment (saml-dev) [inline app.go:85]: does `slog.Error` also exit like `log.Fatalln` does?

File: schedule.go [hunk 2]
```diff
@@ -168,7 +168,7 @@ func runSchedules(a *App) {
 			sched = popSchedule(a)
 		}
 
-		log.Println("Next schedule:", sched.nextRunTime)
+		slog.Info("Next schedule", "next run time", sched.nextRunTime)
```
Comment (saml-dev) [inline schedule.go:171]: nit: swap "next run time" for "start_time"

File: app.go (hunk 1 above)
Comment (jluzny) [reply app.go]: Good point, slog.Error() doesn't exit execution. 
After reading a few discussions and recommendations in books, my summary of the idiomatic way to propagate similar unexpected situations from libraries should still be using Errors (not Fatal or panic). E.g., it allows gracefully catching this in unit/integration tests.
I fixed it according to it.

File: schedule.go (hunk 2 above)
Comment (jluzny) [reply schedule.go]: changed

File: app.go (hunk 1 above)
Comment (saml-dev) [reply app.go]: I agree that is idiomatic in Go, but if there's an error that will prevent GA from working at all I didn't want less experienced Go users to be left wondering why their program was exiting immediately.

However you're probably right that it's best to just follow convention. This solution is good for now, thanks!

**Pull request merged**